src/ThreeEditor/js/object/Object.Grid.ts

```diff
@@ -0,0 +1,115 @@
+//GRID DENSITY
+//WIDTH DENSITY
+//HEIGHT DENSITY
+//DEPTH DENSITY
+
+import { DetectGeometry } from '../../util/Detect/DetectGeometry';
+import { createRowParamNumber, hideUIElement, showUIElement } from '../../util/UiUtils';
+import { SetDetectGeometryCommand } from '../commands/Commands';
+import { Editor } from '../Editor';
+import { UINumber, UIRow } from '../libs/ui';
+import { ObjectAbstract } from './Object.Abstract';
+
+//RADIAL DENSITY
+export class ObjectGrid extends ObjectAbstract {
+	object?: DetectGeometry;
+
+	widthRow: UIRow;
+	width: UINumber;
+
+	heightRow: UIRow;
+	height: UINumber;
+
+	depthRow: UIRow;
+	depth: UINumber;
+
+	radiusRow: UIRow;
+	radius: UINumber;
+
+	constructor(editor: Editor) {
+		super(editor, 'Grid');
+
+		[this.widthRow, this.width] = createRowParamNumber({
+			update: this.update.bind(this),
+			text: `number of bins along X axis`,
+			min: 0,
+			max: 10000,
```

Comment:
```suggestion
			max: 1000000,
```

If this is really max possible number of bins, then I'd suggest going up to 1000000 in all such cases